### PR TITLE
[Progress] Add a progress V2 page with a toggle between v1 and v2

### PR DESCRIPTION
**Pull request by lfryemason**
NOOP without `'progress-table-v2-enabled'` set to `true.

This adds an empty progress V2 page to the progress tab. This page is only shown by toggling the user setting to show the v2 button using an (unstyled) button. 

In order to see the v2 page:
* set `'progress-table-v2-enabled'` to `true` in ruby console
* Go to the progress tab of any section
* Click `toggle V1/V2` button

v1 with `progress-table-v2-enabled` set to `false`
image

v2 with `progress-table-v2-enabled` set to `true`
image

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      </div>
+    );
+  }
+  return (
+    <div>
+      {toggleV1OrV2Button()}
+      <SectionProgress />
+    </div>
+  );
+}
```
Comment (etaderhold) [inline apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx]: Wouldn't something like this work just as well?

```suggestion
  const v2Enabled = DCDO.get('progress-table-v2-enabled', false);
  const isPreferenceSet = showProgressTableV2 !== undefined;
  const displayV2 = v2Enabled && (isPreferenceSet ?
    showProgressTableV2
    DCDO.get('progress-table-v2-default-v2', false)
  );
  
  return (
    <div>
      {v2Enabled && toggleV1OrV2Button()}
      {displayV2 ? (<SectionProgressV2 />) : (<SectionProgress />)}
    </div>
  );
```

File: apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import UserPreferences from '@cdo/apps/lib/util/UserPreferences';
+
+function SectionProgressSelector({
+  showProgressTableV2,
+  setShowProgressTableV2,
+}) {
+  const onShowProgressTableV2Change = useCallback(
+    e => {
+      const shouldShowV2 = !showProgressTableV2;
+      new UserPreferences().setSortByFamilyName(shouldShowV2);
```
Comment (kobryan0619) [inline apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx]: Why does this include `.setSortByFamilyName`?

Comment (lfryemason) [reply apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx]: A copy paste mistake, good catch

File: apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx (hunk 1 above)
Comment (lfryemason) [reply apps/src/templates/sectionProgressV2/SectionProgressSelector.jsx]: I mostly went with your suggestion, however I added an early return for when v2 is completely disabled. Looks much cleaner, thanks!

**Review by etaderhold — APPROVED** [review]

**Pull request merged**